Exercise/src/my_string.cpp

```diff
@@ -173,29 +173,229 @@ Exercise_1::CMyString& Exercise_1::CMyString::operator=(const Exercise_1::CMyStr
 }
 
 // Destructor
-Exercise_1::CMyString::~CMyString()
+MyStructs::CMyString::~CMyString()
 {
 	if(m_szData != nullptr)
 	{
 		delete[] m_szData;
 	}
 }
 
-Exercise_1::CMyString& Exercise_1::CMyString::operator=(const char* cpszCharsSequence)
+void MyStructs::CMyString::Delete(size_t nPosition, size_t nCharactersCount) noexcept
+{
+	if(IsOutOfBounds(nPosition, nCharactersCount))
+	{
+		return;
+	}
+	if(nCharactersCount == size() - 1) 
+	{
+		delete[] m_szData;
+		m_szData = nullptr;
+		m_nSize = 0;
+		return;
+	}
+
+	const size_t cnOldSize = m_nSize;
+	// Calculating new size with getting in mind that user can delete zero terminant
+	m_nSize -= nCharactersCount - IsReachedTerminator(nPosition, nCharactersCount);
+
+
+	char *szOldString = m_szData; 
+	TryToAllocate(m_nSize);
+	if(m_szData == nullptr)
+	{
+		return;
+	}
+	
+	size_t nOldStringPosition = 0;
+	size_t nNewStringPosition = 0;
+
+	// Moving to cnOldSize - 2, to manualy add '\0' in the end in case we deleted it
+	while(nOldStringPosition < cnOldSize - 1) 
+	{
+		if(nOldStringPosition == nPosition)
+		{
+			nOldStringPosition += nCharactersCount;
+			continue;
+		}
+		
+		m_szData[nNewStringPosition] = szOldString[nOldStringPosition];
+		++nOldStringPosition;
+		++nNewStringPosition;
+	}
+
+	m_szData[m_nSize - 1] = '\0';
+
+	delete[] szOldString;
+}
+
+MyStructs::CMyString MyStructs::CMyString::Substring(size_t nPosition, size_t nCharactersCount) const noexcept
+{
+	CMyString substring{nullptr}; 
+	if(nPosition >= m_nSize || nCharactersCount == 0 || 
+		m_nSize - nPosition < nCharactersCount)
+	{
+		return substring;
```

Comment:
You also have to print some error here.
If it is not printed in further functions call

Reply:
Added printf to print error